src/lxc/file_utils.c

```diff
@@ -158,6 +158,21 @@ ssize_t lxc_read_nointr_expect(int fd, void *buf, size_t count, const void *expe
 		return -1;
 	}
 
+	return 0;
+}
+
+ssize_t lxc_read_file_expect(const char *path, void *buf, size_t count, const void *expected_buf)
+{
+	int fd;
+	ssize_t ret;
+
+	fd = open(path, O_RDONLY | O_CLOEXEC);
+	if (fd < 0)
+		return -1;
+
+	ret = lxc_read_nointr_expect(fd, buf, count, expected_buf);
+	close(fd);
```

Comment:
With the cleanup macro above this line needs to be removed.

Reply:
At which point you can remove remove the `ssize_t ret` declaration and return with:
```c
return lxc_read_nointr_expect(...
```

Reply:
@brauner thanks will do